### PR TITLE
[ADO] disallow scan admin controls for non-admin user

### PR DESCRIPTION
**Pull request by arvindsingh-cloud**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 [hunk 1]
```diff
@@ -150,6 +154,8 @@ class SVTResourceResolver: AzSKRoot {
 
     [void] LoadResourcesForScan() {
         
+        $user = [ContextHelper]::GetCurrentSessionUser();
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:157]: move this part to WritePsConsole.ps1

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 [hunk 2]
```diff
@@ -612,4 +634,59 @@ class SVTResourceResolver: AzSKRoot {
             }
         }
     }
+
+    #check for PCA group members
+    [bool] isAdminControlScan()
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:639]: move to administrator helper

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 [hunk 3]
```diff
@@ -612,4 +634,59 @@ class SVTResourceResolver: AzSKRoot {
...
+        $groupsOrgObj = [WebRequestHelper]::InvokePostWebRequest($url,$postbody);
+        
+        if (!$this.ControlSettings) {
+            $this.ControlSettings = [ConfigurationManager]::LoadServerConfigFile("ControlSettings.json");
+        }
+        if ([Helpers]::CheckMember($this.ControlSettings, "AllowAdminControlScanByGroups")) {
+            $allowedAdminGrp = $this.ControlSettings.AllowAdminControlScanByGroups | where { $_.ResourceType -eq "Organization" } | select-object -property GroupNames 
+            $groupsOrgObj = $groupsOrgObj.dataProviders.'ms.vss-admin-web.org-admin-groups-data-provider'.identities | where { $allowedAdminGrp.GroupNames -eq $_.displayName }
+        }
+        else {
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:652]: add comment on why this code was added

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 [hunk 4]
```diff
@@ -612,4 +634,59 @@ class SVTResourceResolver: AzSKRoot {
...
+		$postbody="{'contributionIds':['ms.vss-admin-web.org-admin-groups-data-provider'],'dataProviderContext':{'properties':{'sourcePage':{'url':'https://$($this.organizationName).visualstudio.com/_settings/groups','routeId':'ms.vss-admin-web.collection-admin-hub-route','routeValues':{'adminPivot':'groups','controller':'ContributedPage','action':'Execute'}}}}}" | ConvertFrom-Json
+        $groupsOrgObj = [WebRequestHelper]::InvokePostWebRequest($url,$postbody);
+        
+        if (!$this.ControlSettings) {
+            $this.ControlSettings = [ConfigurationManager]::LoadServerConfigFile("ControlSettings.json");
+        }
+        if ([Helpers]::CheckMember($this.ControlSettings, "AllowAdminControlScanByGroups")) {
+            $allowedAdminGrp = $this.ControlSettings.AllowAdminControlScanByGroups | where { $_.ResourceType -eq "Organization" } | select-object -property GroupNames 
+            $groupsOrgObj = $groupsOrgObj.dataProviders.'ms.vss-admin-web.org-admin-groups-data-provider'.identities | where { $allowedAdminGrp.GroupNames -eq $_.displayName }
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:650]: use -contains

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 [hunk 5]
```diff
@@ -612,4 +634,59 @@ class SVTResourceResolver: AzSKRoot {
...
+        }
+        else {
+            $groupsOrgObj = $groupsOrgObj.dataProviders.'ms.vss-admin-web.org-admin-groups-data-provider'.identities | where { "Project Collection Administrators" -eq $_.displayName }
+        }
+		        
+        $this.isUserPCA = [AdministratorHelper]::GetIsCurrentUserPCA($groupsOrgObj.descriptor, $this.organizationName);
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:656]: add loop for each group

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 (hunk 1 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:157]: moved

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 (hunk 4 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:650]: updated

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 (hunk 5 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:656]: updated added loop for each groups

File: src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1 (hunk 3 above)
Comment (arvindsingh-cloud) [reply src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1:652]: added comment

**Pull request closed without merging**